### PR TITLE
Add Leaderboard run for Keshav Keval

### PR DESCRIPTION
**Pull request by Keshav564**
Final Validation Loss: 3.20313
Link to Wandb report: https://api.wandb.ai/links/keshavpatel564-stanford-university/0mxcgf55

Description: 
* small optimizations such as using register buffers rather than generating tensors in the forward pass etc
* QK Normalization, "smear gate" for attention and FFN with residual stream, embedding tied to final head layer
* torch.compile(mode = 'reduce-overhead') and BF16 for activations
* context-length: 1024, layers: 12, heads: 12, d_model: 768, d_ff: 2048 (SWiGLU activations)
* z-loss added to cross-entropy loss to prevent logit from blowing up (validation loss shown is vanilla CE)
* Batch-size and learning rate tuning: batch-size= 64, max_lr = 2e-3, min_lr = 2e-4

### CODE REVIEW

**Comment by marcelroed** [comment]
Thanks! Can you add a log of what you did as a comment here as well? Otherwise ready to merge.

**Comment by Keshav564** [comment]
Added the description, Thanks!

**Comment by marcelroed** [comment]
Thanks!
